docs/quick_start.md

```diff
@@ -1,127 +1,140 @@
+## Your First Smart Function
 
-Let's see how a smart function looks like with the following example.
-The `request_tez` smart function allows users to request tez from you if asked politely.
+In this example, the smart function takes a HTTP `Request` object and returns a `Response` object
+that sent a tez to the requester if asked politely.
 
-```javascript
-// examples/request_tez.js
-export default async function (request) {
+```typescript
+// <src="examples/get-tez/index.ts">
+
+// 1 tez = 1 million mutez
+const ONE_TEZ = 1000000;
+
+// Maximum amount of tez a requester can receive
+const MAX_TEZ = 10000;
+
+const getRecievedTez = (requester: Address): number => {
+  let receivedTez: number | null = Kv.get(`received/${requester}`);
+  receivedTez = receivedTez === null ? 0 : receivedTez;
+  console.debug(`Requestor already received ${receivedTez} tez`);
+  return receivedTez;
+};
+
+const setRecievedTez = (requester: Address, received: number): void => {
+  Kv.set(`received/${requester}`, received + 1);
+};
+
+const handler = async (request: Request): Promise<Response> => {
   // Extract the requester's address and message from the request
-  const requester_address = request.headers.get("Referer");
+  const requester = request.headers.get("Referer") as Address;
   const { message } = await request.json();
 
-  console.log(`${requester_address} says: ${message}`);
+  console.log(`${requester} says: ${message}`);
 
   // Check if the requester is polite, and decline the request if not
   if (!message.toLowerCase().includes("please")) {
     return new Response("Sorry, I only fulfill polite requests");
   }
 
-  // Check how much tez the requester already received in the Kv store
-  let receivedTez = Kv.get("received/" + requester_address);
-  receivedTez = receivedTez === null ? 0 : receivedTez;
-  console.debug(`Requestor already received ${receivedTez} tez`);
-
   // If the requester already received too much tez, decline the request
-  if (receivedTez >= 10000) {
+  const recievedTez = getRecievedTez(requester);
+  if (recievedTez >= MAX_TEZ) {
     return new Response("Sorry, you already received too much tez");
   }
 
   // Process the request and send the 1 tez = 1 million mutez to the requester if you can
-  if (Ledger.balance(Ledger.selfAddress) > 1000000) {
+  if (Ledger.balance(Ledger.selfAddress) > ONE_TEZ) {
     console.log(
-      `Transferring 1 tez from ${Ledger.selfAddress} to ${requestor}...`,
+      `Transferring 1 tez from ${Ledger.selfAddress} to ${requester}...`,
     );
-    Ledger.transfer(requestor, 1000000);
+    Ledger.transfer(requester, ONE_TEZ);
   } else {
     return new Response(
       "Sorry, I don't have enough tez to fulfill your request",
     );
   }
+  setRecievedTez(requester, recievedTez + 1);
 
-  // Update the amount of tez the requester received in the Kv store
-  Kv.set("received/" + requester_address, received + 1);
-
-  // Pay taxes on the gift by calling a nested smart function.
```

Comment:
I prefer this or something similar to stay in order to easily show how nested calls can be used.